src/AppRouter.js

```diff
@@ -0,0 +1,67 @@
+import { Routes, Route } from "react-router-dom";
+import { RecipeList } from "./modules/recipes/RecipeList";
+import { SearchBar } from "./modules/recipes/SearchBar";
+import { RecipeDetails } from "./modules/recipes/RecipeDetails";
+import { ProtectedRoute } from "./ProtectedRoute";
+import { CreateRecipe } from "./modules/recipes/CreateRecipe";
+import { Login } from "./modules/recipes/Login";
+import { Categories } from "./modules/recipes/Categories"; // Kategorileri import et
+import { CategoryDetails } from "./modules/recipes/CategoryDetails"; // Kategori detaylarını import et
+
+export const AppRouter = ({ recipes, onRecipeClick }) => {
+  return (
+    <Routes>
+      <Route path="/login" element={<Login />} />
+      <Route
+        path="/user"
+        element={
+          <ProtectedRoute>
+            <h1>User page</h1>
+          </ProtectedRoute>
+        }
+      />
+      <Route
+        path="/settings"
+        element={
+          <ProtectedRoute>
+            <h1>Settings page</h1>
+          </ProtectedRoute>
+        }
+      />
+
+      <Route path="/recipes">
+        <Route
+          path="search"
+          element={
+            <>
+              <SearchBar />
+              <RecipeList recipes={recipes} onRecipeClick={onRecipeClick} />
+            </>
+          }
+        />
+        <Route path=":recipeId" element={<RecipeDetails />} />
+        <Route
+          path="create"
+          element={
+            <ProtectedRoute>
+              <CreateRecipe />
+            </ProtectedRoute>
+          }
+        />
+        <Route path="categories" element={<Categories />} />
+        <Route path="category/:category" element={<CategoryDetails />} />
```

Comment:
If you prefer using absolute path instead of a route group, change the path to '/categor**ies**/:category' so that it's consistent with the Categories route.

You'd also need to update Navbar component.